<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MyMainWindow</class>
 <widget class="QMainWindow" name="MyMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1055</width>
    <height>472</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="MyGLWidget" name="widget" native="true">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>0</y>
         <width>561</width>
         <height>431</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>300</width>
         <height>300</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
      </widget>
      <widget class="QListWidget" name="faceListWidget">
       <property name="geometry">
        <rect>
         <x>700</x>
         <y>30</y>
         <width>91</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="QListWidget" name="halfEdgeListWidget">
       <property name="geometry">
        <rect>
         <x>820</x>
         <y>30</y>
         <width>91</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="QListWidget" name="vertexListWidget">
       <property name="geometry">
        <rect>
         <x>940</x>
         <y>30</y>
         <width>91</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="splitFaceButton">
       <property name="geometry">
        <rect>
         <x>690</x>
         <y>270</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Split Face</string>
       </property>
      </widget>
      <widget class="QLabel" name="faceLabel">
       <property name="geometry">
        <rect>
         <x>700</x>
         <y>10</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Faces</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="faceLabel_2">
       <property name="geometry">
        <rect>
         <x>820</x>
         <y>10</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Half-Edges</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="faceLabel_3">
       <property name="geometry">
        <rect>
         <x>940</x>
         <y>10</y>
         <width>92</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Vertices</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton">
       <property name="geometry">
        <rect>
         <x>690</x>
         <y>310</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Add Vertex</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_2">
       <property name="geometry">
        <rect>
         <x>690</x>
         <y>350</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Add Edge</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_3">
       <property name="geometry">
        <rect>
         <x>690</x>
         <y>390</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Remove Vertex</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="redSpinBox">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>290</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="greenSpinBox">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>320</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="blueSpinBox">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>350</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="xSpinBox">
       <property name="geometry">
        <rect>
         <x>960</x>
         <y>290</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="ySpinBox">
       <property name="geometry">
        <rect>
         <x>960</x>
         <y>320</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="zSpinBox">
       <property name="geometry">
        <rect>
         <x>960</x>
         <y>350</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="redLabel">
       <property name="geometry">
        <rect>
         <x>800</x>
         <y>290</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Red</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="greenLabel">
       <property name="geometry">
        <rect>
         <x>800</x>
         <y>320</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Green</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="blueLabel">
       <property name="geometry">
        <rect>
         <x>800</x>
         <y>350</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Blue</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="xLabel">
       <property name="geometry">
        <rect>
         <x>920</x>
         <y>290</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>X:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="yLabel">
       <property name="geometry">
        <rect>
         <x>920</x>
         <y>321</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Y:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="xLabel_2">
       <property name="geometry">
        <rect>
         <x>920</x>
         <y>351</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Z:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="setColorLabel">
       <property name="geometry">
        <rect>
         <x>820</x>
         <y>260</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Set Face Color</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="setColorLabel_2">
       <property name="geometry">
        <rect>
         <x>950</x>
         <y>260</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Move Vertex</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_4">
       <property name="geometry">
        <rect>
         <x>820</x>
         <y>390</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Apply Color</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_5">
       <property name="geometry">
        <rect>
         <x>930</x>
         <y>390</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Apply Position</string>
       </property>
      </widget>
      <widget class="QPushButton" name="playButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>330</y>
         <width>101</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Start Rotation</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pauseButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>380</y>
         <width>101</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Stop Rotation</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_8">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>41</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Up</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_9">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>100</y>
         <width>41</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Down</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_10">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>70</y>
         <width>41</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Right</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>41</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Left</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_6">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>140</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Zoom In</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_7">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>170</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Zoom Out</string>
       </property>
      </widget>
      <widget class="QLabel" name="camControlsLabel">
       <property name="geometry">
        <rect>
         <x>15</x>
         <y>10</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Camera Controls</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="cubeControlsLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>300</y>
         <width>101</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Cube Controls</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1055</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QWidget</extends>
   <header>myglwidget.h</header>
   <container>1</container>
   <slots>
    <slot>setCurrentFace(int)</slot>
    <slot>splitCurrentFace()</slot>
    <slot>startRotation()</slot>
    <slot>stopRotation()</slot>
    <slot>setCurrentVertex(int)</slot>
    <slot>setRed(double)</slot>
    <slot>setGreen(double)</slot>
    <slot>setBlue(double)</slot>
    <slot>changeColor()</slot>
    <slot>setX(double)</slot>
    <slot>setY(double)</slot>
    <slot>setZ(double)</slot>
    <slot>changePos()</slot>
    <slot>setCurrentHalfEdge(int)</slot>
    <slot>insertVertex()</slot>
    <slot>deleteVertex()</slot>
    <slot>insertEdge()</slot>
    <slot>moveUp()</slot>
    <slot>moveRight()</slot>
    <slot>moveLeft()</slot>
    <slot>moveDown()</slot>
    <slot>zoomIn()</slot>
    <slot>zoomOut()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>faceListWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setCurrentFace(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>723</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>splitFaceButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>splitCurrentFace()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>742</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>293</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vertexListWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setCurrentVertex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>985</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setRed(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>859</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>466</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setGreen(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>863</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setBlue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>867</x>
     <y>391</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>409</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>changeColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>855</x>
     <y>438</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>995</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>464</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>982</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>512</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>983</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>481</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>changePos()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>952</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>420</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>halfEdgeListWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setCurrentHalfEdge(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>847</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>insertVertex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>deleteVertex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>757</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>452</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>insertEdge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>740</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>moveUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>moveRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_11</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>moveLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>moveDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>162</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>startRotation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>387</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>stopRotation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
